gradio/events.py

```diff
@@ -398,7 +427,7 @@ def event_trigger(
                 preprocess: If False, will not run preprocessing of component data before running 'fn' (e.g. leaving it as a base64 string if this method is called with the `Image` component).
                 postprocess: If False, will not run postprocessing of component data before returning 'fn' output to the browser.
                 cancels: A list of other events to cancel when this listener is triggered. For example, setting cancels=[click_event] will cancel the click_event, where click_event is the return value of another components .click method. Functions that have not yet run (or generators that are iterating) will be cancelled, but functions that are currently running will be allowed to finish.
-                every: Run this event 'every' number of seconds while the client connection is open. Interpreted in seconds.
+                every: Will be deprecated in favor of gr.Timer. Run this event 'every' number of seconds while the client connection is open. Interpreted in seconds.
```

Comment:
Why does `every` for events behave differently than in components? We should just have the same behavior here, where you can provide a timer or a float?

Reply:
I don't want to encourage the use of `every` added to an event listener which is a very weird API imo, so I'm only supporting floats for the sake of backwards compatibility.

Reply:
Why not? Its quite an intuitive api -- once the event is triggered, it should rerun every X seconds. Its much easier than creating a gr.Timer object and setting it to active, etc.

Reply:
This would be the 4th (!!!) way to support a continuous event, we already have:
1. using a direct timer with a tick listener
2. using every=float in a Component, which creates a timer under the hood
3. using every=Timer in a Component with an existing explicit timer

We keep coming up way too many ways to do the same thing and it makes the API messier and messier.

Reply:
But these are not the same thing. Using `every` in a Component only allows you to rerun a function when the demo first loads. Whereas using `every` in an event listener allows you to rerun a function when an event is triggered.

In fact, I would argue that this would make things *more* consistent. Just like `every` in a Component can be either a float or a Timer, `every` in an event listener can be a float or a Timer

Reply:
Synced with @aliabid94 and I'm convinced that its not a good idea to add yet another approach.